### PR TITLE
fix: alvr_client_core stability issues on smartphone androids

### PR DESCRIPTION
**Pull request by ShootingKing-AM**
- low-latency feature crashes on smartphone android, hence made it as a default-enabled feature https://github.com/PhoneVR-Developers/alvr-cardboard/issues/5

- Building for x86 arch enables faster AVD emulation, wider client_core compatibility
  - Android enforces only x86 emulator images to run on x86 hosts from API 28 - https://stackoverflow.com/a/74819011
    - we can actually stably build for other ABIs as well but waiting for confirmation from https://github.com/googlevr/cardboard/pull/400
    - and fixes from upstream module https://github.com/rust-lang/socket2/issues/433
- some useful logging

future todo: build for remaining two ABIs (ref: https://developer.android.com/ndk/guides/abis)

### CODE REVIEW

File: alvr/client_core/src/platform/android.rs [hunk 1]
```diff
@@ -447,6 +449,8 @@ pub fn video_decoder_split(
             let decoder = Arc::new(FakeThreadSafe(
                 MediaCodec::from_decoder_type(&mime).unwrap(),
             ));
+
+            info!("Using AMediaCoded format:{} ", format);
```
Comment (zarik5) [inline alvr/client_core/src/platform/android.rs:453]: Remove logging

**Review by zarik5 — CHANGES_REQUESTED** [review]
if you add another target (x86_64) expose it as a flag in xtask. You also need to expose it for prepare-deps to download the toochain.

You don't need to expose a low latency flag. you are just modifying settings defaults, which should be accessed from the dashboard directly. The issue is that we didn't implement support for dictionaries yet.

**Comment by ShootingKing-AM** [comment]
> if you add another target (x86_64) expose it as a flag in xtask. 
I felt that it would be redundant, let us build the client_core for widest compatibility and let the client implementation choose which ever ABI they wish to use (which gradle/cmake would automatically pickup). I actually wanted to do `-t arm64-v8a -t armeabi-v7a -t x86_64 -t x86` but was waiting for aforementioned issues to be solve.

> You also need to expose it for prepare-deps to download the toochain.
True, that would be a pr to alvr-cardboard repo

> You don't need to expose a low latency flag. you are just modifying settings defaults, which should be accessed from the dashboard directly. The issue is that we didn't implement support for dictionaries yet.
True, its actually weird that low-latency (well documented flag) crashes with UNKNOWN error. That is default enabled on alvr_client_core, and crashes smartphone androids. I don't think we can request all smartphone android implementations to post installation first goto alvr dashboard and disable that flag :( Thought this would be best wayforward. what do you suggest ?

**Comment by zarik5** [comment]
> True, its actually weird that low-latency (well documented flag) crashes with UNKNOWN error. That is default enabled on alvr, and crashes smartphone androids. I don't think we can request all smartphone android implementations to post installation first goto alvr dashboard and disable that flag :( Thought default-enabled-feature would be best wayforward. what do you suggest ?

This is what I expected the user to do e.g. for an earlier version, where FFR would break the image on the GearVR. But indeed I agree we should apply fixes automatically if possible. The tricky part is detecting which devices support the flag and which don't. Also settings are set on the server side, so the feature flag you put doesn't make sense. My solution would be to add the feature flag on alvr_client_core, and when the client receives the mediacodec options, search for the low-latency entry and remove it conditionally.

**Comment by ShootingKing-AM** [comment]
> This is what I expected the user to do e.g. for an earlier version, where FFR would break the image on the GearVR. But indeed I agree we should apply fixes automatically if possible. The tricky part is detecting which devices support the flag and which don't. Also settings are set on the server side, so the feature flag you put doesn't make sense. My solution would be to add the feature flag on alvr_client_core, and when the client receives the mediacodec options, search for the low-latency entry and remove it conditionally.

This is exactly what I did. `alvr/alvr_client_core` uses `alvr/session` too. When client_core is built its pulling in `session/settings`. And by default, server or default alvr_client_core, when using `alvr/sessions` will have low-latency enabled. Only when `alvr_client_core` is built with `--no-default-features`, `low-latency` will be removed from `alvr/sessions` and pulled into `alvr_client_core`. `alvr/sessions` is rightly being shared between client and server. I have tested this.

**Comment by zarik5** [comment]
No, it's not working correctly. Changing the defaults on the client will do nothing since the settings are pulled from the server. If it works for you it's because you recompile the server without the flag and we don't want to have a separate server build for the cardboard version.

**Comment by ShootingKing-AM** [comment]
- removed default "low-latency" flag, since it crashes on some clients
- Upstream merged fixes (https://github.com/rust-lang/socket2/pull/434), now can build for all ABIs

**Comment by zarik5** [comment]
That's good. So in the end the reason i'm ok with this is because we are still setting the vendor specific qcomm low latency flag. It's still missing the installation of the other targets in prepare-deps.

**Comment by ShootingKing-AM** [comment]
Updated deps ?

**Review by zarik5 — APPROVED** [review]

**Comment by ShootingKing-AM** [comment]
https://github.com/googlevr/cardboard/pull/400 Upstream updated, cardboard-sdk can now be built for x86, x86_64 (instructions are now added to https://developers.google.com/cardboard/develop/c/quickstart#enable_android_studio_x86_emulator). 

Those changes have to be made manually to sdk codebase (to enable building) which are readily available @ https://github.com/ShootingKing-AM/cardboard/tree/phonevr

**Pull request merged**